lib/post_creator.rb

```diff
@@ -414,6 +415,16 @@ def ensure_in_allowed_users
     end
   end
 
+  def make_visible
+    return unless SiteSetting.embed_unlisted?
+    return unless @post.post_number > 1
+    return if @post.topic.visible?
```

Comment:
Can you also assert that the hidden reason == imported_as_unlisted here?

Reply:
I looked at that but the reason is attached to a post and means the first post is hidden, as opposed to the topic being unlisted.